src/extension/content-script/batteries/SoundCloud.ts

```diff
@@ -0,0 +1,102 @@
+import getOriginData from "../originData";
+import { findLightningAddressInText, setLightningData } from "./helpers";
+
+const urlMatcher =
+  /^https:\/\/soundcloud.com\/([^/]+)(\/([^/]+))?(\/([^/]+))?(\/([^/]+))?$/;
+
+function battery(): void {
+  const urlParts = document.location
+    .toString()
+    .replace("https://soundcloud.com", "")
+    .split("/");
+  const username = urlParts[1];
+
+  if (username && !urlParts[2]) {
+    handleProfilePage();
+  } else if (urlParts[2] === "sets" || urlParts[3] === "sets") {
+    handleTrackPage();
+  }
+}
+
+function handleTrackPage() {
+  const description = document.querySelector<HTMLDivElement>(
+    ".truncatedAudioInfo__content .sc-text-body"
+  );
+  if (!description) return;
+
+  const address = findLightningAddressInText(description.innerText);
+  if (!address) return;
+
+  setLightningData([
+    {
+      method: "lnurl",
+      address: address,
+      ...getOriginData(),
+      description:
+        document.querySelector<HTMLDivElement>(
+          ".soundTitle__titleHeroContainer"
+        )?.innerText ?? "",
+      name:
+        document.querySelector<HTMLAnchorElement>(
+          ".userBadge__username .userBadge__usernameLink"
+        )?.innerText ?? "",
+      icon:
+        document
+          .querySelector<HTMLSpanElement>(`.userBadge__avatar span.sc-artwork`)
+          ?.style.backgroundImage.slice(4, -1)
+          .replace(/"/g, "") ?? "",
+    },
+  ]);
+}
+
+function handleProfilePage() {
+  const webProfiles = document.querySelector<HTMLDivElement>(".web-profiles");
+  const linkElement = document.querySelector<HTMLAnchorElement>(
+    ".web-profiles a[href*='getalby.com']"
```

Comment:
This is just a general fallback if i.e. no ⚡ has been found?

Reply:
Yes, not just as a fallback but also because some people prefer to add it directly like this:
![Screenshot from 2022-09-05 12-19-47](https://user-images.githubusercontent.com/64399555/188382768-39591ece-2449-443a-9d83-3f2d9393fbb7.png)
While others add a link:
![Screenshot from 2022-09-05 12-19-21](https://user-images.githubusercontent.com/64399555/188382764-d7af6321-2c65-4980-b7d9-1f9db5fe54d0.png)
Apart from this (web profiles), I've also added the description to be checked for an address if they put it there by any chance!